sound/soc/sof/intel/hda-loader.c

```diff
@@ -518,6 +520,70 @@ int hda_dsp_cl_boot_firmware(struct snd_sof_dev *sdev)
 	return ret;
 }
 
+int hda_dsp_ipc4_load_library(struct snd_sof_dev *sdev,
+			      struct sof_ipc4_fw_library *fw_lib, bool reload)
+{
+	struct sof_intel_hda_dev *hda = sdev->pdata->hw_pdata;
+	struct hdac_ext_stream *hext_stream;
+	struct firmware stripped_firmware;
+	struct sof_ipc4_msg msg = {};
+	struct snd_dma_buffer dmab;
+	int ret, ret1;
+
+	/* IMR booting will restore the libraries as well, skip the loading */
+	if (reload && hda->booted_from_imr)
```

Comment:
don't we need to reload if the firmware crashed? we've done all this logic for the base firmware, not sure what's different about the additional libraries.

We also know that IMR is not preserved when hibernating, so that flags needs to be ignored in at least two cases (crash and hibernate).

Reply:
We will reload since the `booted_from_imr` is going to be false, so we continue to load the library.